presto-hive/src/main/java/com/facebook/presto/hive/HdfsConfigurationUpdater.java

```diff
@@ -51,70 +50,31 @@
     private final Configuration resourcesConfiguration;
     private final HiveCompressionCodec compressionCodec;
     private final int fileSystemMaxCacheSize;
+    private final S3ConfigurationUpdater s3ConfigurationUpdater;
 
-    private final String s3AwsAccessKey;
-    private final String s3AwsSecretKey;
-    private final String s3Endpoint;
-    private final PrestoS3SignerType s3SignerType;
-    private final boolean s3UseInstanceCredentials;
-    private final boolean s3SslEnabled;
-    private final boolean s3SseEnabled;
-    private final PrestoS3SseType s3SseType;
-    private final String s3EncryptionMaterialsProvider;
-    private final String s3KmsKeyId;
-    private final String s3SseKmsKeyId;
-    private final int s3MaxClientRetries;
-    private final int s3MaxErrorRetries;
-    private final Duration s3MaxBackoffTime;
-    private final Duration s3MaxRetryTime;
-    private final Duration s3ConnectTimeout;
-    private final Duration s3SocketTimeout;
-    private final int s3MaxConnections;
-    private final DataSize s3MultipartMinFileSize;
-    private final DataSize s3MultipartMinPartSize;
-    private final File s3StagingDirectory;
-    private final boolean pinS3ClientToCurrentRegion;
-    private final String s3UserAgentPrefix;
+    @VisibleForTesting
+    public HdfsConfigurationUpdater(HiveClientConfig config)
```

Comment:
This might be out of the scope of this PR, but this should be changed to DfsConfigurationUpdater since it deals with more than just HDFS.

Reply:
Agreed. I can do that as a separate PR.